components/infobox/wikis/pubg/infobox_weapon_custom.lua

```diff
@@ -16,58 +17,48 @@ local Weapon = Lua.import('Module:Infobox/Weapon', {requireDevIfEnabled = true})
 
 local Widgets = require('Module:Infobox/Widget/All')
 local Cell = Widgets.Cell
-local Title = Widgets.Title
 local Center = Widgets.Center
+local Title = Widgets.Title
 
-local CustomWeapon = Class.new()
+---@class PubgWeaponInfobox: WeaponInfobox
+local CustomWeapon = Class.new(Weapon)
 local CustomInjector = Class.new(Injector)
 
-local _weapon
-local _args
-
 ---@param frame Frame
 ---@return Html
 function CustomWeapon.run(frame)
-	local weapon = Weapon(frame)
-	_weapon = weapon
-	_args = _weapon.args
-	weapon.createWidgetInjector = CustomWeapon.createWidgetInjector
-	return weapon:createInfobox()
-end
+	local weapon = CustomWeapon(frame)
+	weapon:setWidgetInjector(CustomInjector(weapon))
 
----@return WidgetInjector
-function CustomWeapon:createWidgetInjector()
-	return CustomInjector()
+	return weapon:createInfobox()
 end
 
+---@param id string
 ---@param widgets Widget[]
 ---@return Widget[]
-function CustomInjector:addCustomCells(widgets)
-	local args = _args
-	table.insert(widgets, Cell{
-		name = 'Ammo Type',
-		content = {args.ammotype}
-	})
-	table.insert(widgets, Cell{
-		name = 'Throw Speed',
-		content = {args.throwspeed}
-	})
-	table.insert(widgets, Cell{
-		name = 'Throw Cooldown',
-		content = {args.throwcooldown}
-	})
-	if String.isNotEmpty(args.map1) then
-		local maps = {}
-
-		for _, map in ipairs(_weapon:getAllArgsForBase(args, 'map')) do
-			table.insert(maps, tostring(CustomWeapon:_createNoWrappingSpan(
-						PageLink.makeInternalLink({}, map)
-					)))
+function CustomInjector:parse(id, widgets)
+	local args = self.caller.args
+	if id == 'custom' then
+		Array.appendWith(
+			widgets,
+			Cell{name = 'Ammo Type', content = {args.ammotype}},
+			Cell{name = 'Throw Speed', content = {args.throwspeed}},
+			Cell{name = 'Throw Cooldown', content = {args.throwcooldown}},
+			Cell{name = 'Maps', content = {args.maps}}
```

Comment:
the Maps cell was not in the old one, is it correct that it's supposed to be here?